### ISSUE TITLE
`useless_let_if_seq` should not trigger on multiple assignments

### ISSUE THREAD

**Issue by sgrif** [rust-lang/rust-clippy#3768]
The following code triggers this lint:

```rust
let thing1;
let thing2;

if things.is_empty() {
    thing1 = None;
    thing2 = None;
} else {
    thing1 = Some(stuff1(things));
    thing2 = Some(stuff2(things));
}
```

while this can technically be written as a single expression such as

```rust
let (thing1, thing2) = if things.is_empty() {
    (None, None)
} else {
    (
        Some(stuff1(things)),
        Some(stuff2(things)),
    )
}
```

most folks I've polled agree that it's way harder to read, since the binding is so far removed from the expression its receiving. This lint makes complete sense if there's only a single assignment, but I think it should ignore multiple assignments in the same if statement

**Comment by jfirebaugh** [comment]
This is a duplicate of #2749.
